table/block_based/block_based_table_builder.cc

```diff
@@ -1031,6 +1029,13 @@ void BlockBasedTableBuilder::WriteBlock(const Slice& raw_block_contents,
   if (!ok()) {
     return;
   }
+
+  if (is_data_block && r->table_options.prepopulate_block_cache) {
```

Comment:
Why only data blocks?

Reply:
I will remove the check for all block types to cache

Reply:
I guess it'll be more complicated since each type of block has its own way of construction. There's logic in ```block_based_table_reader.cc``` for constructing different types of blocks, which could be refactored so it can be used in both places. I'm fine if you want to defer it to another PR. For now, pre-populating just data blocks should be beneficial.

Reply:
Make sense. I update it in another PR.